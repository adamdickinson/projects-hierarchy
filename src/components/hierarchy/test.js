import render from "preact-render-to-string"
import { h } from "preact"
import Hierarchy from "./"



describe("Hierarchy component", () => {

  const sampleTree = {
    label: "Sample",
    children: [
      { label: "Example" },
      { label: "Test" }
    ]
  }



  it("should render", () => {
    expect(render(<Hierarchy tree={{ label: "Sample" }} />))
      .toMatchSnapshot()
  })



  it("should render with icon and as clickable", () => {
    expect(render(<Hierarchy tree={{ label: "Sample" }} onClick={jest.fn()} getIcon={() => "icon"} />))
      .toMatchSnapshot()
  })



  it("should render many nodes", () => {
    expect(render(<Hierarchy tree={sampleTree} />))
      .toMatchSnapshot()
  })

})
