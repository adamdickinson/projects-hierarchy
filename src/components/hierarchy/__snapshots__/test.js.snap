// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hierarchy component should render 1`] = `"<svg class=\\"hierarchy\\" viewBox=\\"-100 -100 300 200\\" width=\\"100%\\" height=\\"100%\\"><g><circle class cx=\\"50\\" cy=\\"0\\" r=\\"20\\"></circle><text x=\\"50\\" y=\\"0\\" dy=\\"35\\">Sample</text></g></svg>"`;

exports[`Hierarchy component should render many nodes 1`] = `"<svg class=\\"hierarchy\\" viewBox=\\"-100 -100 400 300\\" width=\\"100%\\" height=\\"100%\\"><line x1=\\"100\\" x2=\\"50\\" y1=\\"0\\" y2=\\"100\\"></line><line x1=\\"100\\" x2=\\"150\\" y1=\\"0\\" y2=\\"100\\"></line><g><circle class cx=\\"100\\" cy=\\"0\\" r=\\"20\\"></circle><text x=\\"100\\" y=\\"0\\" dy=\\"35\\">Sample</text></g><g><circle class cx=\\"50\\" cy=\\"100\\" r=\\"20\\"></circle><text x=\\"50\\" y=\\"100\\" dy=\\"35\\">Example</text></g><g><circle class cx=\\"150\\" cy=\\"100\\" r=\\"20\\"></circle><text x=\\"150\\" y=\\"100\\" dy=\\"35\\">Test</text></g></svg>"`;

exports[`Hierarchy component should render with icon and as clickable 1`] = `"<svg class=\\"hierarchy\\" viewBox=\\"-100 -100 300 200\\" width=\\"100%\\" height=\\"100%\\"><g><circle class=\\"clickable\\" cx=\\"50\\" cy=\\"0\\" r=\\"20\\"></circle><text x=\\"50\\" y=\\"0\\" dy=\\"35\\">Sample</text>icon</g></svg>"`;
